react/hooks/useAppSettings.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from 'react-apollo'
+import { useEffect, useState } from 'react'
+
+import GET_SETTINGS from '../queries/getSettings.graphql'
+
+const DEFAULT_DECIMALS = 2
+const DEFAULT_PRICES_WITH_TAX = false
+
+interface Settings {
+  decimals: number
+  pricesWithTax: boolean
+}
+
+const useAppSettings = (): Settings => {
+  const [settings, setSettings] = useState<Settings>({
+    decimals: DEFAULT_DECIMALS,
+    pricesWithTax: DEFAULT_PRICES_WITH_TAX,
+  })
+
+  const { data, loading } = useQuery(GET_SETTINGS, { ssr: false })
+
+  useEffect(() => {
+    if (!loading) {
+      const { decimals, pricesWithTax } = JSON.parse(data?.appSettings?.message)
+
+      setSettings({ decimals, pricesWithTax })
+    }
+  }, [loading, data])
```

Comment:
Oh, I'm sorry for the early approval. I think this can break if the request fails or if there's nothing on the settings. I think it's better to check it before deconstructing.

```suggestion
  useEffect(() => {
    if (!loading && data?.appSettings?.message) {
      const { decimals, pricesWithTax } = JSON.parse(data.appSettings.message)
      
      if(decimals !== undefined && decimals !== null && pricesWithTax !== null && pricesWithTax !== undefined) {
        setSettings({ decimals, pricesWithTax })
      }
    }
  }, [loading, data, setSettings])
```

Reply:
You're right - deconstructing works regardless of decimals and pricesWithTax being defined, but I added a check after to setSettings state only if they are defined, otherwise the state will stay as it was initially set. This problem appears because the default value in manifest.json for settingsSchema fields only works if you go to the app's settings and press the save changes button once (without changing anything), that's when the default values you have defined in manfiest are applied. Until then they're useless.  Hence why I had to set the initial state to those default values.

Good catch, thanks!